### PR TITLE
fix(frontend): "You are unauthorized to access the requested resource." error message is shown on the Sign-In/ Sign-Up page bb-222

### PR DESCRIPTION
**Pull request by marharita08**
## Summary
Fixed issue: "You are unauthorized to access the requested resource." error message is shown on the Sign-In/ Sign-Up page.

## Updated
- app.tsx: check for token and user existence before call getAuthenticatedUser action

### CODE REVIEW

**Review by ihorLysak — APPROVED** [review]
seems fine to me

File: apps/frontend/src/libs/components/app/app.tsx [hunk 1]
```diff
@@ -10,22 +10,32 @@ import {
...
 		dataStatus: auth.dataStatus,
+		user: auth.user,
 	}));
 
 	const isRoot = pathname === AppRoute.ROOT;
 
 	useEffect(() => {
-		void dispatch(authActions.getAuthenticatedUser());
-	}, [dispatch]);
+		const loadAuthenticatedUser = async (): Promise<void> => {
```
Comment (fshabanov) [inline apps/frontend/src/libs/components/app/app.tsx:29]: Should be handled in slice - if there's no token, we don't make a call

**Review by fshabanov — CHANGES_REQUESTED** [review]

File: apps/frontend/src/modules/auth/slices/actions.ts [hunk 2]
```diff
@@ -39,13 +39,16 @@ const signUp = createAsyncThunk<
...
-	UserDto,
+	null | UserDto,
 	undefined,
 	AsyncThunkConfig
->(`${sliceName}/get-authenticated-user`, async (_, { extra }) => {
+>(`${sliceName}/get-authenticated-user`, async (_, { extra, getState }) => {
 	const { authApi } = extra;
 
-	return await authApi.getAuthenticatedUser();
+	const { user } = getState().auth;
```
Comment (fshabanov) [inline apps/frontend/src/modules/auth/slices/actions.ts:48]: No need to check for a user

File: apps/frontend/src/modules/auth/slices/actions.ts [hunk 3]
```diff
@@ -39,13 +39,16 @@ const signUp = createAsyncThunk<
...
+	const token = await storage.get(StorageKey.TOKEN);
+
+	return token && !user ? await authApi.getAuthenticatedUser() : null;
```
Comment (fshabanov) [inline apps/frontend/src/modules/auth/slices/actions.ts:51]: Avoid complex ternary operators.
```
const hasToken = Boolean(token)

if (!hasToken) {
  return null;
}

return get()
```

**Review by fshabanov — CHANGES_REQUESTED** [review]

**Review by fshabanov — APPROVED** [review]

**Pull request merged**